### PR TITLE
Simplify Nix expressions

### PR DESCRIPTION
**Pull request by expipiplus1**
This uses `developPackage` from `nixpkgs` to generate the derivation

Also add CI for nix build

### CODE REVIEW

**Comment by expipiplus1** [comment]
reopened from https://github.com/lspitzner/brittany/pull/326#event-4092913743

File: default.nix [hunk 1]
```diff
@@ -1,38 +1,18 @@
...
-  license = stdenv.lib.licenses.agpl3;
+
+let
+  haskellPackages = if compiler == null then
+    pkgs.haskellPackages
+  else
+    pkgs.haskell.packages.${compiler};
+
+in haskellPackages.developPackage {
+  name = "";
```
Comment (tfausak) [inline default.nix:15]: Should this have `name = "brittany"`, or does it not matter?

**Review by tfausak — APPROVED** [review]
I am not a Nix wizard but this does indeed look like an improvement. Thanks!

File: default.nix (hunk 1 above)
Comment (expipiplus1) [reply default.nix:15]: It doesn't *really* matter, in this case it'll use the enclosing directory name. However it should be "brittany" to make sure the same derivation is generated regardless of the directory name. Fixed.

**Comment by expipiplus1** [comment]
Thanks!

**Pull request merged**